examples/jsm/renderers/webgpu/WebGPUUniformsGroup.js

```diff
@@ -0,0 +1,138 @@
+import WebGPUBinding from './WebGPUBinding.js';
+
+class WebGPUUniformsGroup extends WebGPUBinding {
+
+	constructor() {
+
+		super();
+
+		this.name = '';
+		this.uniforms = new Map();
+
+		this.update = function () {};
+
+		this.type = 'uniform-buffer';
+		this.visibility = GPUShaderStage.VERTEX;
+
+		this.usage = GPUBufferUsage.UNIFORM | GPUBufferUsage.COPY_DST;
+
+		this.array = null; // set by the renderer
+		this.bufferGPU = null; // set by the renderer
+
+		Object.defineProperty( this, 'isUniformsGroup', { value: true } );
+
+	}
+
+	setName( name ) {
+
+		this.name = name;
+
+	}
+
+	setUniform( name, uniform ) {
+
+		this.uniforms.set( name, uniform );
+
+		return this;
+
+	}
+
+	removeUniform( name ) {
+
+		this.uniforms.delete( name );
+
+		return this;
+
+	}
+
+	setUpdateCallback( callback ) {
+
+		this.update = callback;
+
+		return this;
+
+	}
+
+	copy( source ) {
+
+		this.name = source.name;
+
+		const uniformsSource = source.uniforms;
+
+		this.uniforms.clear();
+
+		for ( const entry of uniformsSource ) {
+
+			this.uniforms.set( ...entry );
+
+		}
+
+		return this;
+
+	}
+
+	clone() {
+
+		return new this.constructor().copy( this );
+
+	}
+
+	getByteLength() {
+
+		let size = 0;
+
+		for ( const uniform of this.uniforms.values() ) {
+
+			size += this.getUniformByteLength( uniform );
+
+		}
+
+		return size;
+
+	}
+
+	getUniformByteLength( uniform ) {
+
+		let size;
+
+		if ( typeof uniform === 'number' ) {
+
+			size = 4;
+
+		} else if ( uniform.isVector2 ) {
+
+			size = 8;
+
+		} else if ( uniform.isVector3 || uniform.isColor ) {
+
+			size = 12;
+
+		} else if ( uniform.isVector4 ) {
+
+			size = 16;
+
+		} else if ( uniform.isMatrix3 ) {
+
+			size = 36;
```

Comment:
If I'm right, it's 48 here?

In my webgpu renderer experience, I found matrix3 seems being 4columns x 3rows in WebGPU memory and fourth columns seem ignored.

https://github.com/takahirox/THREE.WebGPURenderer/blob/master/src/renderers/WebGPURenderer.js#L1066-L1077

Reply:
I did not had a chance to test this yet (the shaders do no yet use a normal matrix yet^^). I have similar code like you in my WebGL2/UBO PR but I wanted to verify first if this sizing is still required.

Reply:
Mine uses normal matrix so it seems to take actually 48 bytes (at last in my demos). But yeah verification should be first. I didn't find sizing of 48 yet neither.